binaries/cli/src/command/inspect/top.rs

```diff
@@ -0,0 +1,392 @@
+            self.sort_ascending = false;
+        }
+        self.sort();
+    }
+
+    fn sort(&mut self) {
+        match self.sort_column {
+            SortColumn::Node => {
+                self.node_stats.sort_by(|a, b| {
+                    let cmp = a.node_id.as_ref().cmp(b.node_id.as_ref());
+                    if self.sort_ascending {
+                        cmp
+                    } else {
+                        cmp.reverse()
+                    }
+                });
+            }
+            SortColumn::Cpu => {
+                self.node_stats.sort_by(|a, b| {
+                    let cmp = a
+                        .cpu_usage
+                        .partial_cmp(&b.cpu_usage)
+                        .unwrap_or(std::cmp::Ordering::Equal);
+                    if self.sort_ascending {
+                        cmp
+                    } else {
+                        cmp.reverse()
+                    }
+                });
+            }
+            SortColumn::Memory => {
+                self.node_stats.sort_by(|a, b| {
+                    let cmp = a
+                        .memory_mb
+                        .partial_cmp(&b.memory_mb)
+                        .unwrap_or(std::cmp::Ordering::Equal);
+                    if self.sort_ascending {
+                        cmp
+                    } else {
+                        cmp.reverse()
+                    }
+                });
+            }
+        }
+    }
+
+    fn update_stats(&mut self, node_infos: Vec<NodeInfo>, system: &System) {
+        self.node_stats.clear();
+
+        for node_info in node_infos {
+            // Try to find process by searching for the node_id in command line
+            let mut pid = None;
+            let mut cpu_usage = 0.0;
+            let mut memory_mb = 0.0;
+
+            // Search for process that might be this node
+            for (process_pid, process) in system.processes() {
+                let cmd = process.cmd()
+                    .iter()
+                    .map(|s| s.to_string_lossy())
+                    .collect::<Vec<_>>()
+                    .join(" ");
+                let name = process.name().to_string_lossy();
+                
+                // Check if this process is related to this node
+                if cmd.contains(node_info.node_id.as_ref()) || name.contains(node_info.node_id.as_ref()) {
+                    pid = Some(process_pid.as_u32());
+                    cpu_usage = process.cpu_usage();
+                    memory_mb = process.memory() as f64 / 1024.0 / 1024.0;
+                    break;
+                }
+            }
+
+            self.node_stats.push(NodeStats {
+                dataflow_id: node_info.dataflow_id,
+                dataflow_name: node_info.dataflow_name.unwrap_or_else(|| "<unnamed>".to_string()),
+                node_id: node_info.node_id,
+                pid,
+                cpu_usage,
+                memory_mb,
```

Comment:
These will always be `None` if the node runs on a different machine than the CLI.

It will also be `None` if the node ID is different from the executable name.